src/people.rs

```diff
@@ -789,6 +861,61 @@ pub trait ContextPeopleExt {
     /// `context.query_people(((Age, 30), (Gender, Female)))`.
     fn query_people<T: Query>(&self, q: T) -> Vec<PersonId>;
     fn match_person<T: Query>(&self, person_id: PersonId, q: T) -> bool;
+    fn get_counts<T: Tabulator, F>(&self, tabulator: &T, print_fn: F)
+    where
+        F: Fn(&Context, &[String], usize);
+}
+
+fn process_indices(
+    context: &Context,
+    remaining_indices: &[&Index],
+    accumulated_values: &mut [IndexValue],
+    display_values: &mut Vec<String>,
+    current_matches: &mut HashSet<PersonId>,
+    intersect: bool,
+    print_fn: &dyn Fn(&Context, &[String], usize),
+) {
+    if remaining_indices.is_empty() {
+        print_fn(context, display_values, current_matches.len());
+        return;
+    }
+
+    if let Some((next_index, rest_indices)) = remaining_indices.split_first() {
+        let lookup = next_index.lookup.as_ref().unwrap();
+
+        // If there is nothing in the index, we don't need to process it
+        if lookup.is_empty() {
+            return;
+        }
+
+        for (value, (people, display)) in lookup {
+            let mut updated_values = accumulated_values.to_owned();
+            updated_values.push(value.clone());
```

Comment:
It looks to me like you don't use `accumulated_values` except to maintain the stack here, so can you just remove it?